pipeline/api-service/base/kustomization.yaml

```diff
@@ -1,10 +1,12 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
 resources:
 - deployment.yaml
 - role-binding.yaml
 - role.yaml
 - sa.yaml
 - service.yaml
-
+namespace: kubeflow
```

Comment:
This is a place I think a top level kustomization.yaml would really be useful. Removing it and going for something custom seems like anti-pattern. 
Or would kfctl take care of setting the namespace from the app.yaml and pushing it down to the individual kustomization files?

Reply:
yes, kfctl is setting the namespace (overriding the one above) in a 'component' level kustomization.yaml. So the tree below:
```
api-service
└── base
    ├── deployment.yaml
    ├── kustomization.yaml
    ├── role-binding.yaml
    ├── role.yaml
    ├── sa.yaml
    └── service.yaml
```
would then look like
```
api-service
├── base
│   ├── deployment.yaml
│   ├── kustomization.yaml
│   ├── role-binding.yaml
│   ├── role.yaml
│   ├── sa.yaml
│   └── service.yaml
└── kustomization.yaml
```

The new kustomization.yaml would look like
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
bases:
- base/kustomization.yaml
namespace: kubeflow
```

Reply:
this is done during generation as just to override certain values like namespace. Though if we're using multiple overlays we would need to specify the child kustomization attributes. 
Kustomize makes this pretty easy by using their ResAccumulator class which allows MergeResourcesWithOverride and other things.